python/pip_install/pip_repository.bzl

```diff
@@ -431,14 +429,44 @@ def _pip_repository_impl(rctx):
     content = rctx.read(requirements_txt)
     parsed_requirements_txt = parse_requirements(content)
 
-    packages = [(_clean_pkg_name(name), requirement) for name, requirement in parsed_requirements_txt.requirements]
+    # Apply name normalizations to the composite libs def once
```

Comment:
Doc recommendations:

How and why are we normalizing this data?  What are the composite libs?